dpdispatcher/submission.py

```diff
@@ -519,17 +519,18 @@ def handle_unexpected_job_state(self):
             raise RuntimeError("job_state for job {job} is unknown".format(job=self))
 
         if job_state == JobStatus.terminated:
-            dlog.info(f"job: {self.job_hash} {self.job_id} terminated; restarting job")
-            if self.fail_count > 3:
-                raise RuntimeError("job:job {job} failed 3 times".format(job=self))
             self.fail_count += 1
+            dlog.info(f"job: {self.job_hash} {self.job_id} terminated;"
+                "fail_cout is {self.fail_count}; resubmitting job")
```

Comment:
I think this "resubmitting" should only be printed when fail_count<=3?

Reply:
Yes, you are right.